tests/functional/aws-node-sdk/test/bucket/listingCornerCases.js

```diff
@@ -475,4 +475,56 @@ describe('Listing corner cases tests', () => {
             }
         });
     });
+
+    it('should not list DeleteMarkers for version suspended buckets', done => {
+        const obj = { name: 'testDeleteMarker.txt', value: 'foo' };
+        const bucketName = `bucket-test-delete-markers-not-listed${Date.now()}`;
+        let objectCount = 0;
+        return async.waterfall([
+            next => s3.createBucket({ Bucket: bucketName }, err => next(err)),
+            next => {
+                const params = {
+                    Bucket: bucketName,
+                    VersioningConfiguration: {
+                        Status: 'Suspended',
+                    },
+                };
+                return s3.putBucketVersioning(params, err =>
+                    next(err));
+            },
+            next => s3.putObject({
+                    Bucket: bucketName,
+                    Key: obj.name,
+                    Body: obj.value,
+                }, err =>
+                next(err)),
+            next => s3.listObjectsV2({ Bucket: bucketName },
+                (err, res) => {
+                    if (err) {
+                        return next(err);
+                    }
+                    objectCount = res.Contents.length;
```

Comment:
here we should check that we did indeed receive the object (testDeleteMarker.txt) : e.g. by going through the result and checking the key is found...